### PR TITLE
MTDSA-3690 - brought forward losses - connector

### PR DESCRIPTION
**Pull request by jeremystone**
- test that des headers passed through for existing connectors
- implement test and connector
- remove unnecessary explicitly passed implicits

### CODE REVIEW

File: app/v1/connectors/DesConnector.scala [hunk 1]
```diff
@@ -17,47 +17,59 @@
...
 
   def createBFLoss(createBFLossRequest: CreateBFLossRequest)(implicit hc: HeaderCarrier,
-                                   ec: ExecutionContext): Future[DesOutcome[CreateBFLossResponse]] = {
+                                                             ec: ExecutionContext): Future[DesOutcome[CreateBFLossResponse]] = {
 
-    val nino    = createBFLossRequest.nino.nino
+    val nino = createBFLossRequest.nino.nino
 
-    val url = s"${appConfig.desBaseUrl}/income-tax/brought-forward-losses/$nino"
+    def doIt(implicit hc: HeaderCarrier) =
```
Comment (james-work-account) [inline app/v1/connectors/DesConnector.scala]: Why was this moved into its own function?

Comment (james-work-account) [inline app/v1/connectors/DesConnector.scala]: Also `DOTI` would probably be more appropriate 👀

File: app/v1/connectors/DesConnector.scala [hunk 2]
```diff
@@ -17,47 +17,59 @@
...
-                                                             ec: ExecutionContext): Future[DesOutcome[AmendBFLossResponse]] = {
+                                                          ec: ExecutionContext): Future[DesOutcome[AmendBFLossResponse]] = {
 
-    val nino    = amendBFLossRequest.nino.nino
-    val lossId  = amendBFLossRequest.lossId
+    val nino   = amendBFLossRequest.nino.nino
+    val lossId = amendBFLossRequest.lossId
 
-    val url = s"${appConfig.desBaseUrl}/income-tax/brought-forward-losses/$nino/$lossId"
+    def doIt(implicit hc: HeaderCarrier) =
```
Comment (james-work-account) [inline app/v1/connectors/DesConnector.scala]: See above

File: app/v1/connectors/DesConnector.scala [hunk 3]
```diff
@@ -17,47 +17,59 @@
...
-    http.PUT(url, amendBFLossRequest.amendBroughtForwardLoss)(AmendBFLoss.writes,
-      StandardDesHttpParser.reads[AmendBFLossResponse], desHeaderCarrier, implicitly)
+    doIt(desHeaderCarrier)
   }
 
+  def deleteBFLoss(request: DeleteBFLossRequest)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[DesOutcome[Unit]] = {
+    val nino   = request.nino.nino
+    val lossId = request.lossId
+
+    def doIt(implicit hc: HeaderCarrier) =
```
Comment (james-work-account) [inline app/v1/connectors/DesConnector.scala]: See above

File: app/v1/connectors/DesConnector.scala (hunk 1 above)
Comment (jeremystone) [reply app/v1/connectors/DesConnector.scala]: So that we can pass an explicit HeaderCarrier without the need to explicitly supply all of the other implicits every time.
